packages/cf-deploy-config-inquirer/src/prompts/prompts.ts

```diff
@@ -144,3 +182,33 @@ export async function getQuestions(
 
     return questions;
 }
+
+/**
+ * Retrieves a list of deployment questions based on the application root and destination name prompt option.
+ *
+ * @param {CfDeployConfigRouterPromptOptions} promptOptions - The configuration options for prompting during cf target deployment.
+ * @param {Logger} [log] - The logger instance to use for logging.
+ * @returns {CfDeployConfigRouterQuestions[]} Returns an array of questions related to cf deployment configuration.
+ */
+export async function getQuestionsWithRouterOptions(
```

Comment:
Should `getQuestions` not be updated rather than adding this new function?
Deprecate `addManagedAppRouter` and use `showRouterOptions` ?

Reply:
Soo this is introducing a change in behaviour and returning a new type i.e. routerType. I think it's better to use `deprecated` instead.